### PR TITLE
Fix hook to work with astropy version 5.0, which requires metadata

### LINKED ISSUE #379: astropy 5.0 not working with pyinstaller
## Description of the issue
pyinstaller seems to not work with the latest version of astropy (5.0 and above)

### Context information (for bug reports)

* Output of `pyinstaller --version`: 4.9
* Version of Python: 3.9.7
* Platform: macOSS Big Sur 11.6.2
* How you installed Python: conda
* Did you also try this on another platform? Does it work there? I did not try this.

### A minimal example program which shows the error

app.py:
```
from PySide6 import QtWidgets
from astropy.time import Time
import sys

class MainWindow(QtWidgets.QMainWindow):

    def __init__(self):
        super().__init__()

        Time('1999-01-01T00:00:00.123456789')
        self.setWindowTitle("Hello World")
        l = QtWidgets.QLabel("My simple app.")
        l.setMargin(10)
        self.setCentralWidget(l)
        self.show()

if __name__ == '__main__':
    app = QtWidgets.QApplication(sys.argv)
    w = MainWindow()
    app.exec()
```

First, install requisites using `pip install pyinstaller=4.9 pyside6'

Then, let's show that this works with the previous version of Astropy:
1. Install astropy 4.3 using `pip install astropy==4.3`
2. Build the package using `pyinstaller app.py`.
3. Run the bundled application using `./dist/app/app`
4. Success!

Then, let's show that this fails with the current version of Astropy:
1. Install latest version of astropy using `pip install astropy==5.0.1`
2. Build the package using `pyinstaller app.py`.
3. Run the bundled application using `./dist/app/app`
4. Failure. See stacktrace below.

### Stacktrace / full error message

```
Traceback (most recent call last):
  File "astropy/utils/introspection.py", line 157, in minversion
  File "importlib/metadata.py", line 551, in version
  File "importlib/metadata.py", line 524, in distribution
  File "importlib/metadata.py", line 187, in from_name
importlib.metadata.PackageNotFoundError: numpy

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "app.py", line 2, in <module>
    from astropy.time import Time
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "PyInstaller/loader/pyimod03_importers.py", line 495, in exec_module
  File "astropy/time/__init__.py", line 19, in <module>
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "PyInstaller/loader/pyimod03_importers.py", line 495, in exec_module
  File "astropy/time/formats.py", line 16, in <module>
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "PyInstaller/loader/pyimod03_importers.py", line 495, in exec_module
  File "astropy/units/__init__.py", line 17, in <module>
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "PyInstaller/loader/pyimod03_importers.py", line 495, in exec_module
  File "astropy/units/quantity.py", line 25, in <module>
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "PyInstaller/loader/pyimod03_importers.py", line 495, in exec_module
  File "astropy/utils/compat/__init__.py", line 19, in <module>
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "PyInstaller/loader/pyimod03_importers.py", line 495, in exec_module
  File "astropy/utils/compat/numpycompat.py", line 14, in <module>
  File "astropy/utils/decorators.py", line 546, in wrapper
  File "astropy/utils/introspection.py", line 164, in minversion
KeyError: 'numpy'
[16774] Failed to execute script 'app' due to unhandled exception!
```

### PR DESCRIPTION
**Pull request by EfremBraun**
Fixes #379 and https://github.com/astropy/astropy/issues/12837

### CODE REVIEW

File: src/_pyinstaller_hooks_contrib/hooks/stdhooks/hook-astropy.py [hunk 1]
```diff
@@ -31,6 +31,10 @@
 
 datas += ply_files
 
+# Astropy queries metadata to get version information.
```
Comment (rokm) [inline src/_pyinstaller_hooks_contrib/hooks/stdhooks/hook-astropy.py:34]: Should this be conditional?
```python
from PyInstaller.utils.hooks import is_module_satisfies

[...]

if is_module_satisfies('astropy >= 5.0'):
   # collect metadata
```

Comment (EfremBraun) [reply src/_pyinstaller_hooks_contrib/hooks/stdhooks/hook-astropy.py:34]: Yes. The issue does not appear prior.

**Comment by pllim** [comment]
Awesome! Please let me know when pyinstaller is released with this patch so I know to check the CI downstream. 🙇‍♀️

**Review by bwoodsend — APPROVED** [review]
Looks good. Just need a [news entry](https://github.com/pyinstaller/pyinstaller-hooks-contrib/blob/master/news/README.txt).

**Comment by pllim** [comment]
I dunno... I have to ask @astrofrog

**Comment by EfremBraun** [comment]
> Looks good. Just need a [news entry](https://github.com/pyinstaller/pyinstaller-hooks-contrib/blob/master/news/README.txt).

Done.

**Comment by astrofrog-conda-forge** [comment]
If keeping the hook in the astropy repo itself is the preferred option, that seems reasonable, though I'd also be fine with testing things here. What is the best option? I haven't had a chance to familiarize myself with how a hook would be added directly to the package, but in any case we will need a fix here until a new version of astropy is released, right?

**Comment by astrofrog** [comment]
If keeping the hook in the astropy repo itself is the preferred option, that seems reasonable, though I'd also be fine with testing things here. What is the best option? I haven't had a chance to familiarize myself with how a hook would be added directly to the package, but in any case we will need a fix here until a new version of astropy is released, right?
If anyone has time to help add the hook to the astropy repo if that is the preferred way forward, that would be great, as I probably won't have time in the near future

**Comment by bwoodsend** [comment]
It's released.

**Pull request merged**